presto-main/src/main/java/io/prestosql/sql/gen/SqlTypeBytecodeExpression.java

```diff
@@ -40,14 +40,24 @@ public static SqlTypeBytecodeExpression constantType(CallSiteBinder callSiteBind
         return new SqlTypeBytecodeExpression(type, binding, BOOTSTRAP_METHOD);
     }
 
+    private static String generateName(Type type)
+    {
+        String name = type.getTypeSignature().toString().replaceAll("\\W+", "_");
+        if (name.length() > 20) {
+            // Use type base to reduce the identifier size in generated code
+            name = type.getTypeSignature().getBase();
+        }
+
+        return name;
```

Comment:
I think we need the regex replace on the base name also.  Maybe this:
```java
        String name = type.getTypeSignature().toString();
        if (name.length() > 20) {
            // Use type base to reduce the identifier size in generated code
            name = type.getTypeSignature().getBase();
        }
        return name.replaceAll("\\W+", "_");
```

Reply:
Updated the diff. Sorry for the late response, I missed the update email.